### PR TITLE
Interval points output processes

### PR DESCRIPTION
**Pull request by frawahl**
**Description**
I had to implement these changes to make my output work, but I am not sure it is valuable for Kratos in general.
I added an interval to the **PointOutputProcess** and its subclasses and also put the search for the respective point in the respective model part in a new method, which gets called at the beginning of the interval and not the initialization anymore. 
This way, the model part doesn't need to exist when the simulation is started, but when the interval starts. In my case that was necessary in order to get output from the model part created by the **ApplyEmbeddedSkinVisualizationProcess**, which is not yet available at initialization.

**Changelog**
- **PointOutputProcess**: Added check for interval to _ExecuteFinalizeSolutionStep(self)_, so output is only written if time is inside interval
- **PointOutputProcess**: Added _self.point_ as class variable and put code that searches for point from ___init_ into new method __SearchPoint(self)_
-  **PointOutputProcess**: ___SearchPoint()_ is now still called in ___init_ if process was restarted, but called in _ExecuteInitializeSolutionStep(self)_ once at the beginning of the interval for a new process
- Added entry for interval to ___init_ of **MultiplePointsOutputProcess**, **LineOutputProcess** and **CSVPointsOutputProcess**
- Created reference file for unittest of interval and added test and interval entry to **test_processes.py**

### CODE REVIEW

**Comment by philbucher** [comment]
I have some comments, will add them this week

File: kratos/python_scripts/csv_points_output_process.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 # other imports
 from KratosMultiphysics.multiple_points_output_process import MultiplePointsOutputProcess
-import csv, os
+import csv
```
Comment (philbucher) [inline kratos/python_scripts/csv_points_output_process.py:6]: +1 for cleaning unused imports

File: kratos/python_scripts/point_output_process.py [hunk 2]
```diff
@@ -89,23 +103,73 @@ def ExecuteInitialize(self):
...
+            self.__SearchPoint()
+
+    def ExecuteBeforeSolutionLoop(self):
+        pass
+
+    def ExecuteInitializeSolutionStep(self):
+        time = self.model_part.ProcessInfo[KratosMultiphysics.TIME]
+
+        # search for point, moved here, so search happens at start time of the time interval
+        if time >= self.interval[0] and not self.search_done:
```
Comment (philbucher) [inline kratos/python_scripts/point_output_process.py]: can you pleas use the `IntervalUtility` for this?

File: kratos/python_scripts/point_output_process.py [hunk 3]
```diff
@@ -89,23 +103,73 @@ def ExecuteInitialize(self):
...
+            self.search_done = True
+            self.__SearchPoint()
+
+    def ExecuteFinalizeSolutionStep(self):
+        time = self.model_part.ProcessInfo[KratosMultiphysics.TIME]
+
+        if(time >= self.interval[0] and time <= self.interval[1]):
```
Comment (philbucher) [inline kratos/python_scripts/point_output_process.py]: same

**Review by philbucher — COMMENTED** [review]
minor comments otherwise noce job :+1:

File: kratos/python_scripts/csv_points_output_process.py (hunk 1 above)
Comment (frawahl) [reply kratos/python_scripts/csv_points_output_process.py:6]: thanks :)

File: kratos/python_scripts/point_output_process.py (hunk 2 above)
Comment (frawahl) [reply kratos/python_scripts/point_output_process.py]: sure! I hadn't seen it before, it's nice.

File: kratos/python_scripts/point_output_process.py [hunk 4]
```diff
@@ -65,7 +70,7 @@ def ExecuteInitialize(self):
         point_position = self.params["position"].GetVector()
         if point_position.Size() != 3:
             raise Exception('The position has to be provided with 3 coordinates!')
-        point = KratosMultiphysics.Point(point_position[0],
+        self.point = KratosMultiphysics.Point(point_position[0],
                                          point_position[1],
                                          point_position[2])
```
Comment (rubenzorrilla) [inline kratos/python_scripts/point_output_process.py:75]: I think this can be initialized in the constructor. Up to you.

**Review by rubenzorrilla — APPROVED** [review]

**Review by philbucher — APPROVED** [review]

**Pull request merged**